db/migrate/20251113124140_add_from_template_permissions_to_roles_with_add_permissions.rb

```diff
@@ -0,0 +1,52 @@
+# frozen_string_literal: true
+
+#-- copyright
+# OpenProject is an open source project management software.
+# Copyright (C) the OpenProject GmbH
+#
+# This program is free software; you can redistribute it and/or
+# modify it under the terms of the GNU General Public License version 3.
+#
+# OpenProject is a fork of ChiliProject, which is a fork of Redmine. The copyright follows:
+# Copyright (C) 2006-2013 Jean-Philippe Lang
+# Copyright (C) 2010-2013 the ChiliProject Team
+#
+# This program is free software; you can redistribute it and/or
+# modify it under the terms of the GNU General Public License
+# as published by the Free Software Foundation; either version 2
+# of the License, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program; if not, write to the Free Software
+# Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+#
+# See COPYRIGHT and LICENSE files for more details.
+#++
+
+require Rails.root.join("db/migrate/migration_utils/permission_adder")
+
+class AddFromTemplatePermissionsToRolesWithAddPermissions < ActiveRecord::Migration[8.0]
+  def up
+    ::Migration::MigrationUtils::PermissionAdder.add(:add_project, :add_project_from_template)
+    ::Migration::MigrationUtils::PermissionAdder.add(:add_programs, :add_programs_from_template)
+    ::Migration::MigrationUtils::PermissionAdder.add(:add_portfolios, :add_portfolios_from_template)
+  end
+
+  def down
+    # Remove the permissions that were added
+    remove_permission_from_roles(:add_project_from_template)
+    remove_permission_from_roles(:add_programs_from_template)
+    remove_permission_from_roles(:add_portfolios_from_template)
+  end
+
+  private
+
+  def remove_permission_from_roles(permission)
+    RolePermission.where(permission: permission.to_s).delete_all
```

Comment:
`.to_s` not needed and `delete_by` shortcut can be used:
```suggestion
    RolePermission.delete_by(permission:)
```